Gradient Descent/gradient.cpp

```diff
@@ -0,0 +1,156 @@
+#include <iostream>
+#include <vector>
+#include <math.h> 
+#include <numeric>
+#include <cstdlib>
+
+using namespace std;
+
+const double epsilon = pow(10,-6); // Fix a small epsilon for finite difference scheme to calculate grad
+
+// This is the function to be minimised. At the moment it is f(x) = x^2
+double objfun(vector<double>& x){               // Takes a vector x...
+    double output_val = 0.0;
+    for(int i=0 ; i < x.size(); i++){
+        output_val += pow(x[i],2);}             //... and computes x_i^2 for each coordinate...
+    return output_val;                      //... and returns vector of calculation.
+}
```

Comment:
True in 1D, actually calculating x^T x (inner product) in higher dimensions (\sum_{i=1}^N x_i^2 0 for x \in R^N